### PR TITLE
feat(ecs): log retention for FireLensLogDriver

### PR DESCRIPTION
**Pull request by sakurai-ryo**
This PR resolves the missing `logs:PutRetentionPolicy` permission issue when using `FireLensLogDriver` with CloudWatch Logs.

### Description
When using `FireLensLogDriver` to send logs to CloudWatch Logs, we can specify the retention period for newly created Log Groups by specifying `log_retention_days` in the `FireLensLogDriverProps.options`.
https://docs.fluentbit.io/manual/pipeline/outputs/cloudwatch#configuration-parameters
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs.FireLensLogDriverProps.html#options

If you have not added a FluentBit container, CDK will automatically add it to the task definition, and the IAM permissions required for this are added to the task role.
https://github.com/aws/aws-cdk/blob/db22b85c9b2a853aa2f830c182a340f0bcf95d1a/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L816
https://github.com/aws/aws-cdk/blob/db22b85c9b2a853aa2f830c182a340f0bcf95d1a/packages/aws-cdk-lib/aws-ecs/lib/firelens-log-router.ts#L170

While `FireLensLogDriver` allows specifying `log_retention_days` for Log Groups, FluentBit cannot set the retention period due to the absence of the `logs:PutRetentionPolicy` policy. Consequently, it results in an `AccessDeniedException`.
To address this, the PR adds the necessary `logs:PutRetentionPolicy` permission to the task role when `log_retention_days` is set in `FireLensLogDriverProps.options`, ensuring FluentBit has the required permissions to set the retention period for Log Groups.

Relates to #28258

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by lpizzinidev — COMMENTED** [review]
Looks good to me! 👍
I think that this should be a `feat`: `feat(ecs): support log retention for FireLensLogDriver`.
You could add some documentation about the newly supported option [here](https://github.com/aws/aws-cdk/blob/93681e07ad19c08f60eb2ee5748a2d55c6d2bc45/packages/aws-cdk-lib/aws-ecs/README.md?plain=1#L1070).

**Comment by sakurai-ryo** [comment]
@lpizzinidev 
Thanks again for your review 👍 
I added docs to the README.

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 1]
```diff
@@ -1107,6 +1107,31 @@ taskDefinition.addContainer('TheContainer', {
...
+      log_stream_prefix: 'from-fluent-bit',
+      auto_create_group: 'true',
+      log_retention_days: '1',
+    },
+  }),
+});
+```
+
+See the [Fluent Bit CloudWatch Configuration Parameters](https://docs.fluentbit.io/manual/pipeline/outputs/cloudwatch#configuration-parameters)
+for more information about `log_retention_days`.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ecs/README.md:1133]:
```suggestion
> Visit [Fluent Bit CloudWatch Configuration Parameters](https://docs.fluentbit.io/manual/pipeline/outputs/cloudwatch#configuration-parameters)
for more details.
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Small documentation nit.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Review by sumupitchayan — DISMISSED** [review]

**Review by kaizencc — APPROVED** [review]

**Pull request merged**